optimum/habana/transformers/modeling_attn_mask_utils.py

```diff
@@ -50,16 +50,14 @@ def _make_causal_mask(
 
         # add lower triangular sliding window mask if necessary
         if sliding_window is not None:
-            diagonal = past_key_values_length - sliding_window + 1
+            diagonal = past_key_values_length - sliding_window - 1
 
-            # Replace triu with below
+            # Replace tril with below
             row_indices = torch.arange(mask.size(0), device=mask.device).view(-1, 1)  # Reshape to column vector
             col_indices = torch.arange(mask.size(1), device=mask.device)
-            context_mask = 1 - (col_indices >= row_indices + diagonal).int().expand_as(
-                mask
-            )  # Expand to match mask shape
+            context_mask = (col_indices <= row_indices + diagonal).bool().expand_as(mask)  # Expand to match mask shape
```

Comment:
we have a .bool() here instead of .int(). We dont support bool, not sure if this will cause a fallback to cpu, etc